openedx2zim/html_processor.py

```diff
@@ -22,11 +22,14 @@ def download_and_get_filename(
             with_ext: ensure that downloaded file has the given extension
             filter_ext: download only if the file to download has an extension in this list """
 
-        if with_ext:
-            ext = with_ext
+        server_path = pathlib.Path(urllib.parse.urlparse(src).path)
+        ext = with_ext if with_ext else server_path.suffix
+
+        if server_path.suffix != "":
+            filename = server_path.with_suffix(ext).name
         else:
-            ext = pathlib.Path(urllib.parse.urlparse(src).path).suffix
-        filename = hashlib.sha256(str(src).encode("utf-8")).hexdigest() + ext
+            filename = hashlib.sha256(str(src).encode("utf-8")).hexdigest() + ext
```

Comment:
Not that important given the volume but we could have used a non-cryptographic/faster hash here.

Reply:
Used md5